test/CMakeLists.txt

```diff
@@ -44,6 +44,44 @@ option( MIOPEN_TEST_MLIR "Add tests for MLIR -- EXPERIMENTAL" ${MIOPEN_USE_MLIR}
 option( WORKAROUND_ISSUE_898 "" ON)
 option( WORKAROUND_ISSUE_936 "" ON)
 
+# Detect GPU type for testing
+if(NOT (MIOPEN_TEST_VEGA OR MIOPEN_TEST_GFX908))
+    find_program(ROCMINFO
+        NAMES rocminfo
+        PATHS
+            /opt/rocm
+            ${CMAKE_INSTALL_PREFIX}
+        PATH_SUFFIXES
+            /bin
+    )
+    message(STATUS "rocminfo utility: ${ROCMINFO}")
+    if(ROCMINFO)
+        execute_process (
+            COMMAND ${ROCMINFO}
+            COMMAND grep -i -E "(gfx900)|(gfx906)"
```

Comment:
Why is the grep necessary? Why not just do `if(MATCHES)` on the output?

Reply:
Who knows? Maybe I simply like grep more than all other tools))

Reply:
...but capturing once + if(MATCHES) provide more compact code. Resolved.